csep/core/regions.py

```diff
@@ -1331,3 +1341,313 @@ def california_quadtree_region(magnitudes=None, name="california-quadtree"):
     california_region = QuadtreeGrid2D.from_quadkeys(qk, magnitudes=magnitudes, name=name)
     return california_region
 
+#--------------- Forecast mapping from one grid to another ----------
+def geographical_area_from_qk(quadk):
+    """
+    Wrapper around function geographical_area_from_bounds
+    """
+    bounds = tile_bounds(quadk)
+    return geographical_area_from_bounds(bounds[0], bounds[1], bounds[2], bounds[3])
+
+
+def tile_bounds(quad_cell_id):
+    """
+    It takes in a single Quadkkey and returns lat,longs of two diagonal corners using mercantile
+    Parameters
+    ----------
+    quad_cell_id : Stirng
+        Quad key of a cell.
+
+    Returns
+    -------
+    bounds : Mercantile object
+        Latitude and Longitude of bottom left AND top right corners.
+
+    """
+    # t = Tile.from_quad_tree('{}'.format(quad_cell_id)) # tile from a quad tree repr string #This will change for Mercantile
+    # bounds = t.bounds
+    bounds = mercantile.bounds(mercantile.quadkey_to_tile(quad_cell_id))
+    return [bounds.west, bounds.south, bounds.east, bounds.north]
+
+
+def create_polygon(fg):
+    """
+    Required for parallel processing
+    """
+    return shapely.geometry.Polygon([(fg[0], fg[1]), (fg[2], fg[1]), (fg[2], fg[3]), (fg[0], fg[3])])
+
+
+def calc_cell_area(cell):
+    """
+    Required for parallel processing
+    """
+    return geographical_area_from_bounds(cell[0], cell[1], cell[2], cell[3])
+
+
+def _map_overlapping_cells(fcst_grid_poly, fcst_cell_area, fcst_rate_poly, target_poly):  # ,
+    """
+    ------Computationally Expensive --- To be used only for Cells that do not directly conside with target poly -----
+    Note: ALERT !!! This function uses 3 global variables, i.e. fcst_grid_poly, fcst_cell_area, fcst_rate_poly
```

Comment:
how are these global variables? do they need to be defined as `global` in the calling script?

Reply:
These functions are updating them inside wrapper function. It should not be a problem outside this funciton.

Reply:
so they aren't global but defined within the scope of the calling function as a closure?